dipy/core/sphere.py

```diff
@@ -496,6 +496,43 @@ def disperse_charges(hemi, iters, const=.2):
     return HemiSphere(xyz=charges), potential
 
 
+def fibonacci_sphere(n_points, randomize=True):
+    """
+    Generate points on the surface of a sphere using Fibonacci Spiral.
+
+    Parameters
+    ----------
+    n_points : int
+        The number of points to generate on the sphere surface.
+    randomize : bool, optional
+        If True, randomize the starting point on the sphere. Default is True.
+
+    Returns
+    -------
+    points : ndarray
+        An array of 3D points representing coordinates on the sphere surface.
+
+    """
+    random_shift = 0
+    if randomize:
+        random_generator = np.random.default_rng()
+        random_shift = random_generator.integers(0, n_points)
+
+    indices = np.arange(n_points)
+    offset = 2.0 / n_points
```

Comment:
is there a minimal number of points  needed? Can you add a small tests (`if`) to check that and avoid invalid value like a division by zero.

```python
if nb_points < 0:
   raise ValueError("my message")
```